flow/compositor_context.cc

```diff
@@ -187,9 +187,9 @@ void CompositorContext::ScopedFrame::PaintLayerTreeImpeller(
     flutter::LayerTree& layer_tree,
     std::optional<SkRect> clip_rect,
     bool ignore_raster_cache) {
-  if (canvas() && clip_rect) {
-    canvas()->Translate(-clip_rect->x(), -clip_rect->y());
-  }
+  // if (canvas() && clip_rect) {
```

Comment:
Need to figure out why this was firing. Possibly we're turning off partial repaint based on thread merging when it should be based on platform view state.

Reply:
Done

Reply:
We need better way to disable partial repaint. I think instead of
```dart
 bool force_full_repaint =
          external_view_embedder_ &&
          (!raster_thread_merger_ || raster_thread_merger_->IsMerged());
```
we should reset damage and clipRect inside `CompositorContext::ScopedFrame::Raster` after `Preroll` when we know that there are platform view present.

Reply:
That would be after we preroll with damage based cull rect though?

Reply:
Yeah, fair point. Though I'm not 100% sure where the cullrect is used during preroll, the actual culling is done through `quickReject` during `Paint`. I think long time ago it was done in preroll setting a flag on layer.

Maybe better to mark entire frame dirty in `PlatformViewLayer::Diff`. Just need to make sure it behaves correctly when diff is skipped for retained layers. I'll look into this as a separate PR.

Reply:
https://github.com/flutter/engine/pull/54219